applications/GeoMechanicsApplication/custom_elements/geo_linear_truss_element_base.cpp

```diff
@@ -263,7 +263,20 @@ double GeoTrussElementLinearBase<TDim,TNumNodes>::CalculateLinearStrain()
     this->CreateTransformationMatrix(transformation_matrix);
 
     current_disp = prod(Matrix(trans(transformation_matrix)),current_disp);
-    const double length_0 = GeoStructuralMechanicsElementUtilities::CalculateReferenceLength3D2N(*this);
+
+    double length_0;
+    if constexpr (TDim == 2)
+    {
+        length_0 = GeoStructuralMechanicsElementUtilities::CalculateReferenceLength2D2N(*this);
+    } else if constexpr (TDim == 3)
+    {
+        length_0 = GeoStructuralMechanicsElementUtilities::CalculateReferenceLength3D2N(*this);
+    }
+    else // 1D

```

Comment:
I am not sure under which circumstance the "else" point will be touched as the only templates that appear to be possible are:

template class GeoTrussElementLinearBase<2,2>;
template class GeoTrussElementLinearBase<3,2>;

Reply:
yes you are right, this can be removed

Reply:
In my opinion, else should be there when we have if{} else if{} . When 1D is not the case, the code should give an error message to prevent any problem in future if the element is extended to 1D geomtery.

Reply:
@Vahid-Galavi Agreed, was thinking the same, @aronnoordam perhaps a not implemented error. Problem with leaving it in, is that a casual developer would assume it's implemented for 1D ..... Of course feel free to implement in 1D ;)